scripts/wmt/README.md

```diff
@@ -1,49 +1,115 @@
 
 ```bash
-pip install OpenNMT-tf[tensorflow_gpu]>=1.10.0
+apt-get install cmake pkg-config libprotobuf9v5 protobuf-compiler libprotobuf-dev libgoogle-perftools-dev
+apt-get install libprotobuf9v5
+git clone https://github.com/google/sentencepiece.git
+cd sentencepiece
+mkdir build
+cd build
+cmake ..
+make -j $(nproc)
+make install
+ldconfig -v
 ```
 
-## Steps
+If you installed SentencePiece in a custom location, change the `SP_PATH` variable in the scripts.
+
+
+### Data loading and preparation with SentencePiece
 
-### Data preparation
+The `prepare_data.sh` script automatically downloads the default datasets (`commoncrawl.de-en`, `europarl-v7.de-en`, `news-commentary-v11.de-en`, `newstest2014-deen` and `newstest2017-ende`) using wget, extracts the files, and tidies the folders for you. If you would like to change the task or datasets used, look at the links in the file header to find their download paths. Changing the source and target language is also possible as long as the associated datasets are available. Be sure to read the logic of the file cleaning to see what has to be adapted to new data.
 
-Before running the script, look at the links in the file header to download the datasets. Depending on the task, you may need to change the filenames and the folders paths.
+The script will also train a SentencePiece model using the same source and target vocabulary. It will tokenize the dataset and prepare the train/valid/test files. The generated files will go into a `data/` directory.
+
+Run these steps using the command:
 
 ```bash
-./prepare_data.sh /data/wmt/
+./prepare_data.sh raw_data
 ```
 
-where `/data/wmt/` contains the raw parallel datasets.
+where `raw_data` is the name of the folder that the raw parallel datasets will be downloaded into.
+
 
-The script will train a SentencePiece model using the same source and target vocabulary. It will tokenize the dataset and prepare the train/valid/test files. A new directory `data/` will contain the generated files.
+### Training the transformer model
 
-### Training
+Now that the train/valid/test files have been generated, kick off a training run using the following command. We recommend training on 4 GPUs to get the best performance.
 
-We recommend training on 4 GPUs to get the best performance:
 
 ```bash
 CUDA_VISIBLE_DEVICES=0,1,2,3 ./run_wmt_ende.sh
 ```
 
+If you have e.g. 3 GPUs, change the `--num-gpus=4` in the `run_wmt_ende.sh` script to `--num-gpus=3` and set `CUDA_VISIBLE_DEVICES=0,1,2`.
+
 Or if you have only 1 GPU, run the dedicated script:
 
 ```bash
 CUDA_VISIBLE_DEVICES=0 ./run_wmt_ende_1gpu.sh
 ```
 
-### Translate
+The training logs printed to STDOUT include the values:
++ `words_per_sec/features`: number of input words processed per second (where features = words in the training set)
```

Comment:
Change to "number of source tokens processed per second" and remove the content in parenthesis ("source" is clear enough).